src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ConfigurationGeneral.xaml

```diff
@@ -42,7 +42,7 @@
     </StringProperty>    
     <StringProperty Name="TargetFrameworkMoniker" DisplayName="Target Framework Moniker">
         <StringProperty.DataSource>
-            <DataSource Persistence="ProjectFileWithInterception" PersistedName="TargetFrameworkMoniker" HasConfigurationCondition="False" SourceOfDefaultValue="AfterContext" />
```

Comment:
What did you take this approach instead of just replacing existing version? When we would ever want to use ProjectFileWithInterception?

Reply:
The existing version is still needed for TargetFrameworkMonikers, PreBuildEvent and PostBuildEvent as they use things that are not available when opting in to CPSes new code path.

Originally https://github.com/dotnet/project-system/pull/3464 tried to change the existing implementation, but it got reverted because of the errors from those properties (well, specifically from TargetFrameworkMonikers). In the DevDiv work item linked above there is a comment from Lifeng (sorry, can't link directly to comments it seems) suggesting that it would be beneficial to add a new provider and move over whichever properties can be moved, to remove UI delays, so thats what I did.

Reply:
It's not clear to me when or when we wouldn't use this implementation. Those properties that ring a bell as to why they would be special. How do we make a decision to use the data source or not?

Reply:
I don't claim to know the full context of the CPS changes, but anecdotally the new provider can be used if the value provider can do its work based on the unevaluated value (eg OutputType converting strings to ints), but the existing provider if more there are more complex requirements (eg TargetFrameworkMonikers which iterates through all configurations).

Realistically for new properties being added, it probably wouldn't matter which provider is used unless the property value is requested enough to warrant the new provider for performance reasons (values are read from the snapshot, instead of under a lock).